ext/sockets/sockaddr_conv.c

```diff
@@ -60,7 +60,8 @@ int php_set_inet6_addr(struct sockaddr_in6 *sin6, char *string, php_socket *php_
 
 	}
 
-	if (scope++) {
+	if (scope) {
+		scope++;
```

Comment:
I think this would be more elegant if moved after the declarations, before is_numeric_string.